models/webhook/hooktask.go

```diff
@@ -100,7 +100,6 @@ type HookResponse struct {
 // HookTask represents a hook task.
 type HookTask struct {
 	ID              int64 `xorm:"pk autoincr"`
-	RepoID          int64 `xorm:"INDEX"`
```

Comment:
A webhook with an event will produce one or many tasks. i.e. An org webhook will generate a hook task for every organization repository.

Reply:
So, which problem are you seeing?

Reply:
You will not know which repository the task belongs to.

Reply:
Why do you need know that? Do you mistake this struct with `Webhook`?

Reply:
No. I'm not. Gitea have repository level webhook, organization webhook and system webhook.

I have said if the webhook is a system one or an organization one. Then it will create a hook task for every repo in the scope. This column `RepoID` is what's the real the webhook to this repo. It must not be removed.

Reply:
But the `RepoID` column is not referenced anywhere but in `services/webhook/deliver.go@241`. Organization and system webhooks still work with my changes.

Reply:
You can find the reference here https://github.com/go-gitea/gitea/blob/main/models/webhook/hooktask.go#L210

Reply:
That method is not needed and I removed it.

Reply:
I think we need list org and system webhook hook task on repository. Then we need the column.

Reply:
I don't think that's a good idea because it could leak private data to the repository owner. If Gitea would display a triggered system/org hook on the repo page, the repo owner could see 1) if there are system/org hooks present and 2) collect the target infos (endpoint, service, ...). Sure, you could filter the listed webhooks with the users access rights but that's already done if you simply use the system/org webhook view.

In my opinion the normal usecase for the webhook history is: you look at it if there is an error. If everything works you are not interested in that information. And in the error case you can simply view the webhook history at the system/org page. So displaying the system/org webhooks at repo level does not have much value.

Reply:
But even if the organization owner want to know the failure webhook is belongs to which repository?

Reply:
If the owner doesn't know which repo fails he simply opens the webhook history in the org webhook view and get's it from the data. How would it help if it is displayed at repo page? If the owner doesn't know the failing repo he can't open that repos settings.

The reason why a webhook fails will (always?) be the receiving end and not the source repo.

Reply:
Even if you opens the webhook history in the org webhook view you cannot get to know which repository failed if you removed the `repo_id` column from hook task. Nobody can know.

Reply:
Ok, I will add it back and mark it as unused.

Reply:
> Ok, I will add it back and mark it as unused.

Thanks. And then I think maybe you can also revert https://github.com/go-gitea/gitea/pull/17940/files#diff-1851799b06733db4df3ec74385c1e8850ee5aedee70b8b55366910d22725eea8L918 ?